doc/source/admin/container_resolvers.rst

```diff
@@ -0,0 +1,276 @@
+During the container resolution the resolvers specified in this
+list are sequentially applied stopping at the first resolver that
+yields a container description. 
+
+The YAML defines a list of container resolvers and their properties.
+This might look as follows:
+
+.. code-block:: yaml
+
+   - type: "cached_explicit_singularity"
+   - type: "cached_mulled_singularity"
+   - type: "mulled_singularity"
+     auto_install: False
+
+would define a list of three container resolvers, and the last setting an option
+of this resolver.
+
+The available container resolver types are described below.
+The properties depend on the resolver type (see documentation
+of the resolvers). Properties that do not apply to the resolver
+are ignored.
+
+.. note::
+
+   Its's important to note that similarities in the names not necessarily
+   imply any similarity in the function of the container resolvers.
+
+Main resolver types:
+--------------------
+
+The main types of container resolvers follow this naming scheme: 
+``[cached_][explicit,mulled][_singularity]``. That is
+
+- a container resolver is either ``explicit`` or ``mulled``
+- cached if it is prefixed with ``cached_`` and non-cached otherwise. 
+- yield a container discription suitable for singularity if
+  suffixed by ``_singularity`` and docker otherwise.
+
+1. docker vs singularity
+
+Galaxy can execute tools in containers using  ``docker`` or ``singularity``.
+The corresponding container resolvers yield container descriptions suitable
+for the corresponding "executor".
+Note that for the execution with ``singularity`` Galaxy relies mostly on
+docker containers that are either executed directly or are converted
+to singularity images (except for explicit container requirements of
+``type="singularity"``).
+
+There are important differences between Galaxy's cached docker and singularity
+container resolvers. The caching mechanism essentially executes a
+``docker pull`` or ``singularity pull``, respectively. For docker this creates
+an entry in the docker image cache (on the local node) whereas for
+singularity an image file is created in the specified ``cache_directory``.
+On distributed systems ``cache_directory`` needs to be accessible on all
+compute nodes.
+
+.. note::
+
+   Using a cached docker resolver has no additional value on distributed compute
+   systems since the cache is only available locally. 
+   Therefore an additional ``docker inspect ... ; [ $? -ne 0 ] && docker pull ...``
+   command is used in each job script.
+   For distributed compute systems built in techniques of docker may be useful:
+   https://docs.docker.com/registry/recipes/mirror/.
+
+
+2. mulled vs explict
+
+Mulled container resolvers apply for requirements defined by tools that are
+a set of packages:
+
+.. code-block:: xml
+
+  <requirements>
+      <requirement type="package" version="0.5">foo</requirement>
+      <requirement type="package" version="1.0">bar</requirement>
+  </requirements>
```

Comment:
Maybe a mention could be added here to the ability to create multi tool containers via the biocontainers UI / pull request mechanism? Or to subscribe your tool list to the CI that monitors tools that get added to the toolshed for this kind of tools?

Reply:
I wrote a bit here already https://github.com/galaxyproject/galaxy/blob/889f7f7413d0b3609b6829279f678e8a0e06e63f/doc/source/admin/container_resolvers.rst#building-resolver-types

It misses at the moment that also bioconda pushes to biocontainers. Maybe it should be moved to a separate subsection and also quay.org and namespaces need to be mentioned. Will put it on my todo list.